sdk/core/azure-core/azure/core/pipeline/_tools.py

```diff
@@ -33,38 +39,26 @@ def await_result(func, *args, **kwargs):
         )
     return result
 
-def to_rest_request(pipeline_transport_request):
-    from ..rest import HttpRequest as RestHttpRequest
-    return RestHttpRequest(
-        method=pipeline_transport_request.method,
-        url=pipeline_transport_request.url,
-        headers=pipeline_transport_request.headers,
-        files=pipeline_transport_request.files,
-        data=pipeline_transport_request.data
-    )
+def is_rest(obj):
+    # type: (Any) -> bool
+    """Return whether a request or a response is a rest request / response.
 
-def to_rest_response(pipeline_transport_response):
-    from .transport._requests_basic import RequestsTransportResponse
-    from ..rest._requests_basic import RestRequestsTransportResponse
-    if isinstance(pipeline_transport_response, RequestsTransportResponse):
-        response_type = RestRequestsTransportResponse
-    else:
-        raise ValueError("Unknown transport response")
-    response = response_type(
-        request=to_rest_request(pipeline_transport_response.request),
-        internal_response=pipeline_transport_response.internal_response,
-        block_size=pipeline_transport_response.block_size
-    )
-    return response
+    Checking whether the response has the object content can sometimes result
+    in a ResponseNotRead error if you're checking the value on a response
+    that has not been read in yet. To get around this, we also have added
+    a check for is_stream_consumed, which is an exclusive property on our new responses.
+    """
+    return hasattr(obj, "is_stream_consumed") or hasattr(obj, "content")
```

Comment:
Why "or"?

Reply:
Basically, if you check `hasattr(obj, content)` on an unread response, it will raise the `ResponseNotReadError`, so I added the `is_stream_consumed` check first, because it's a response-specific property, and will not raise when checked ever

Reply:
Why not hasattr(obj, "is_stream_consumed") only?

Reply:
I use this check for requests and responses, and the requests don't have the attr "is_stream_consumed". Ideally I would just keep it at "content", but sadly can't because of the ResponseNotReadError, and if it throws it, know that it's a rest response, but swallow the error. I also thought about catching ResponseNotReadError, but I decided it best to not trigger an error if I didn't need to. Lmk your thoughts though, def open to any of your improvements

Reply:
Get it. 

So hasattr(obj, "is_stream_consumed") is for response and hasattr(obj, "content") is for request.

Thanks for the clarification.

Reply:
thanks!